Packs/MapPattern/Scripts/MapPattern/MapPattern.py

```diff
@@ -424,22 +425,27 @@ def main():
                         fields_comp_mode=fields_comp_mode,
                         wildcards=wildcards,
                         regex_flags=regex_flags)
+
+        matched = False
         if fields_comp_mode:
             if isinstance(value, dict):
-                value, _ = tr.translate_fields(
+                value, matched = tr.translate_fields(
                     obj_value=value,
                     field_mapping=mappings,
                     priority=priority,
                     algorithm=algorithm)
         else:
             if not isinstance(value, (dict, list)):
-                value, _ = tr.translate(
+                value, matched = tr.translate(
                     source=value,
                     pattern_mapping=mappings,
                     priority=priority,
                     algorithm=algorithm)
-    except Exception as err:
-        return_error(err)
+        if default_value and not matched:
+            value = default_value
+    except Exception:
+        # Don't return an error by return_error() as this is transformer.
+        raise
```

Comment:
Let's use `raise DemistoException` here to inform the users what were the problems.